wgpu-hal/src/vulkan/mod.rs

```diff
@@ -793,19 +1058,20 @@ impl crate::Queue for Queue {
     ) -> Result<(), crate::SurfaceError> {
         let mut swapchain = surface.swapchain.write();
         let ssc = swapchain.as_mut().unwrap();
+        let mut swapchain_semaphores = texture.surface_semaphores.lock();
+
+        // debug_assert_eq!(
+        //     Arc::as_ptr(&texture.surface_semaphores),
+        //     Arc::as_ptr(&ssc.surface_semaphores[ssc.next_semaphore_index]),
+        //     "Trying to use a surface texture that does not belong to the current swapchain."
+        // );
```

Comment:
This should either be uncommented or removed, but we shouldn't leave it in limbo like this.